src/specializations/BezierCurve.jl

```diff
@@ -80,9 +80,8 @@ function integral(
         FP::Type{T} = Float64,
         alg::Meshes.BezierEvalMethod = Meshes.Horner()
 ) where {F <: Function, T <: AbstractFloat}
-    len = length(curve)
     point(t) = curve(t, alg)
-    integrand(t) = len * f(point(t[1]))
+    integrand(t) = f(point(only(t))) *  norm(derivative(curve, only(t)))
 
     # HCubature doesn't support functions that output Unitful Quantity types
     # Establish the units that are output by f
```

Comment:
Unrelated to this PR, but I just saw: Shouldn't the line below read `testpoint_parametriccoord = zeros(FP, 1)`. I mean, technically it doesn't matter in this case because we apply `only` to it anyway (or before `t[1]`), but the `paramdim` is 1. So the testpoint should also only have length 1, right?

Reply:
Oh, I see it does matter. `only` doesn't work if the vector has more than one element (even if they are the same).

Reply:
I fixed that in 25eda748bc00056353d90fb5929f0b63d97ee84d.

Reply:
For some `only` isn't working here as expected, and I'm a bit puzzled as to why.